lessons/07-mcp/README.md

```diff
@@ -0,0 +1,482 @@
+       "type":"object",
+       "properties":{
+          "a":{
+            "type":"number"
+          },
+          "b":{
+            "type":"number"
+          }},
+          "required":["a","b"],
+          "additionalProperties":false,
+          "$schema":"http://json-schema.org/draft-07/schema#"
+      }
+  }
+```
+
+which means, if you have the tools `add` and `subtract`, your response looks like so:
+
+**You**: "Ok, good, I guess I can store that in a variable and then call the tool I want?"
+
+**Time Beetle**: "Exactly! Let's show calling a tool next:"
+
+```typescript
+// List tools
+const { tools } = await client.listTools();
+
+const addTool = tools[0]; // Assuming the first tool is "add"
+const subtractTool = tools[1]; // Assuming the second tool is "subtract"
+
+// Call a tool
+const result = await client.callTool({
+  name: addTool.name,
+  arguments: {
+    a: 5,
+    b: 10
+  }
+});
+```
+
+**You**: "Ok, makes sense. And what about resources?"
+
+**Time Beetle**: "Resources are similar, but you need to provide the resource URI and any parameters it requires. Here's an example:
+
+```typescript
+let resourceUri = "greeting://John"; // Replace with the actual resource URI
+
+// call resource
+// Read a resource
+const name = "John Doe"; // Replace with the actual name
+
+const resourceResult = await client.readResource({
+  uri: `greeting:///${name}`,
+});
+
+console.log(resourceResult); // Hi John Doe!
+```
+
+**You**: "I get it nice, although, you're thinking to yourself, I bet we can improve this somehow? Time Beetle, we can improve this right? 
+
+**Time Beetle**: Yes, but first things first, let's make Scipio happy. We need to have a chat with a Ms Lamarr on our next stop.
+
+## Assignment - creating an inventory and tools management system for Scipio
+
+**Scipio**: "You know, I have so many parchments and scrolls lying around. I could really use a tool to help me organize them. I also have this abacus that I use for calculations. Can you help me create a tool to manage my inventory?"
+
+Here's a complete inventory of my scrolls:
+
+- Scroll of Tactics.
+- Scroll of food supply.
+
+Oh, here's a Taxation scroll. Taxation is big thing, the Empire wants their cut of everything we acquire. Have you solved this in your time, don't answer that, I don't want to know.
+
+The abacus tool, I need it to support the following operations:
+
+- Add: Add two numbers together.
+- Subtract: Subtract one number from another.
+- Multiply: Multiply two numbers together.
+- Divide: Divide one number by another.
+
+## Solution
+
+[Solution](/lessons/07-mcp/solution/README.md)
+
+## Knowledge Check
+
+**Question**: What are the different types of resources an MCP Server can expose?
+
+A. Tools and Web endpoints
+
+B. Tools and Resources
+
+C. Tools, Resources and Prompts
+
+[Solution Quiz](/lessons/07-mcp/solution/solution-quiz.md)
+
+**Question**: What are some ways you can test your MCP Server?
+
+A. Inspector
+
+B. Written client
```

Comment:
Maybe change this to something like:

A custom MCP client

A "Written client" isn't super clear as to what that actually is.